### PR TITLE
18544 va confirmed appointment details

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
Add VA (in-person and phone) confirmed appointment details page styling and interaction.

## Testing done
Unit Testing

## Screenshots
Screen Shot 2021-01-28 at 6 17 16 PM

## Acceptance criteria
- [ ] All unit test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx [hunk 1]
```diff
@@ -1,186 +1,239 @@
...
-
-import RequestedAppointmentDetailsPage from '../../../appointment-list/components/RequestedAppointmentDetailsPage';
-import { renderWithStoreAndRouter } from '../../mocks/setup';
-import {
-  getVAFacilityMock,
-  getVARequestMock,
-  getMessageMock,
-} from '../../mocks/v0';
-import { FETCH_STATUS } from '../../../utils/constants';
+import { setFetchJSONFailure } from 'platform/testing/unit/helpers';
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx:5]: Are the changes in this unit test changes you made? They seem unrelated to the page you're adding.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx [hunk 2]
```diff
@@ -437,4 +442,154 @@ describe('VAOS <AppointmentsPageV2>', () => {
...
+          mentalHealthClinic: '307-778-7349',
+          enrollmentCoordinator: '307-778-7550 x7579',
+        },
+        uniqueId: '442GC',
+      },
+    };
+    mockFacilitiesFetch('vha_442GC', [facility]);
+
+    const screen = renderWithStoreAndRouter(
+      <MemoryRouter initialEntries={['/', url]}>
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:516]: There's already a Router component rendered in the `renderWithStoreAndRouter`. I think you could remove `MemoryRouter` here and possibly render `AppointmentListSection` from `src/applications/vaos/appointment-list/index.jsx`, so you don't have to put the individual routes in here. Mocking up specific routes like this makes the test easier to break later if we change how the routing works.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx [hunk 3]
```diff
@@ -437,4 +442,154 @@ describe('VAOS <AppointmentsPageV2>', () => {
     ).to.be.ok;
     expect(screen.getByRole('link', { name: 'Request appointment' }));
   });
+
+  it('should navigate to confirmed appointments detail page', async () => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:446]: I see why you put this test in here, since it starts on an appointment list page and does test the link to the details page, but it does seem like it mostly tests the rendering in the details page itself. What do you think about putting it a separate test file, like Narin's tests for the request detail page?

File: src/applications/vaos/components/VAFacilityLocation.jsx [hunk 4]
```diff
@@ -7,7 +7,8 @@ export default function VAFacilityLocation({
   facility,
   facilityName,
   facilityId,
-  isHomepageRefresh,
+  isV2,
```
Comment (jbalboni) [inline src/applications/vaos/components/VAFacilityLocation.jsx:10]: This seems like a bad merge, I think Narin changed `isV2` to `isHomepageRefresh`

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 5]
```diff
@@ -422,6 +435,35 @@ export function fetchRequestDetails(id) {
...
+      type: FETCH_REQUEST_DETAILS,
+    });
+
+    if (appointment) {
+      dispatch({ type: FETCH_REQUEST_DETAILS_SUCCEEDED, appointment, id });
+    } else {
+      // TODO: fetch single appointment
+    }
+
+    // TODO: Why???
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:457]: This section isn't necessary for your page, since VA appointments don't have messages.

File: src/applications/vaos/components/FacilityAddress.jsx [hunk 6]
```diff
@@ -7,6 +7,8 @@ export default function FacilityAddress({
   name,
   facility,
   showDirectionsLink,
+  isV2,
```
Comment (jbalboni) [inline src/applications/vaos/components/FacilityAddress.jsx:10]: I think Narin switched these props to isHomepageRefresh elsewhere, which seems more clear to me.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  const { id } = useParams();
+  const history = useHistory();
+
+  const appointment = appointmentDetails?.[id];
+
+  useEffect(() => {
+    const status = confirmedStatus === FETCH_STATUS.succeeded;
+
+    if (!status) {
+      history.push('/va');
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:52]: I think the goal of this in Narin's PR was to go back to one of the list pages when the item wasn't already fetched, so this url would need to be one of those list page routes.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+  }
+
+  if (!appointment) {
+    return null;
+  }
+
+  const facilityId = getVAAppointmentLocationId(appointment);
+  const facility = facilityData?.[facilityId];
+
+  const isCC = appointment?.vaos.isCommunityCare;
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:77]: This looks like a duplicate of `isCommunityCare`.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+  const isExpressCare = appointment?.vaos.isExpressCare;
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:78]: I think you assume this is always false, since these are only requests, not confirmed items.

File: src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx [hunk 10]
```diff
@@ -31,7 +31,7 @@ function parseFakeFHIRId(id) {
 }
 
 function RequestedAppointmentDetailsPage({
-  appointment,
+  appointmentDetails,
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:34]: I think the changed in this file might also not be intentional

File: src/applications/vaos/appointment-list/index.jsx [hunk 11]
```diff
@@ -1,18 +1,32 @@
...
+          component={() => (
+            <PageLayout>
+              <ConfirmedAppointmentDetailsPage showCancelButton />
+            </PageLayout>
+          )}
+        />
+      )}
+      {featureHomepageRefresh && (
+        <Route
+          path="/request/:id"
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/index.jsx:29]: I think this was changed to `requests`

File: src/applications/vaos/appointment-list/redux/reducer.js [hunk 12]
```diff
@@ -150,22 +152,18 @@ export default function appointmentsReducer(state = initialState, action) {
...
-      const appointmentDetails = { ...state.appointmentDetails };
-
-      appointmentDetails[action.id] = action.request;
-
+    case (FETCH_REQUEST_DETAILS_SUCCEEDED, FETCH_CONFIRMED_DETAILS_SUCCEEDED): {
       return {
         ...state,
-        currentAppointment: action.request,
-        appointmentDetails,
+        appointmentDetails: {
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/reducer.js:164]: I think you need a `...state.appointmentDetails,` in this object, so that we don't remove any other appointment detail items that might be there.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx [hunk 13]
```diff
@@ -437,4 +442,154 @@ describe('VAOS <AppointmentsPageV2>', () => {
...
+    // Verify page content...
+    expect(
+      await screen.findByRole('heading', {
+        level: 1,
+        name: new RegExp(moment().format('dddd, MMMM DD, YYYY'), 'i'),
+      }),
+    ).to.be.ok;
+
+    // NOTE: This 2nd 'await' is needed!!! Seems like we need to wait for
+    // conditionals to process in addition to any async code.
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:546]: The facility details are fetched asynchronously, so this page could render once with appointment info but without the facility details. That's why the second await is needed.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+  const isCommunityCare = appointment?.vaos.isCommunityCare;
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:79]: Community care is kind of tricky here, since the majority of CC appointments won't use this page, they get data from a different service. But we do have the potential for CC appointments from the VA appointments service (it's turned off in prod, though). 

I'm not positive about the best way to handle this, but I would lean toward assuming `isCommunityCare` is always false here and handling both types of CC appointments in the CC appointment details page.

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 15]
```diff
@@ -422,6 +435,35 @@ export function fetchRequestDetails(id) {
...
+  return async (dispatch, getState) => {
+    const state = getState();
+    const { appointmentDetails } = state.appointments;
+    const futureAppointments = selectFutureAppointments(state);
+    const appointment =
+      appointmentDetails[id] ||
+      futureAppointments?.find(appt => appt.id === id);
+
+    dispatch({
+      type: FETCH_REQUEST_DETAILS,
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:448]: Should these two dispatches in here use the new action names you created?

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx [hunk 16]
```diff
@@ -437,4 +442,154 @@ describe('VAOS <AppointmentsPageV2>', () => {
...
+            city: 'Fort Collins',
+            state: 'CO',
+            address1: '2509 Research Boulevard',
+            address2: null,
+            address3: null,
+          },
+        },
+        id: 'vha_442GC',
+        name: 'Fort Collins VA Clinic',
+        phone: {
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:501]: I think you could leave the phone numbers you're not using out of the data here, to shorten the test up a little

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+        />
+      </h1>
+
+      {!!facility &&
+        !isCC &&
+        !isExpressCare && (
+          <>
+            <div className="vads-u-font-size--sm vads-u-font-weight--normal vads-u-font-family--sans">
+              <span>
+                <b>{header}</b>
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:111]: Can we use `<strong>` instead of `<b>` here? Also, is there a reason we're combining vads-u-font-weight--normal with a bolding element? Looks like we're trying mimic a heading here. Should we be using a heading?

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+              />
+              <a href="#">Print</a>
+            </div>
+
+            <div className="vads-u-margin-top--2 vaos-appts__block-label">
+              <i
+                aria-hidden="true"
+                className="fas fa-clock vads-u-margin-right--1"
+              />
+              <a href="#">Reschedule</a>
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:160]: Needs to be either a full URL in the href or a button

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+                  </span>
+                </button>
+              </div>
+            )}
+          </>
+        )}
+
+      <div className="vads-u-margin-top--3 vaos-appts__block-label">
+        <Link to="/">
+          <button className="usa-button vads-u-margin-top--2">
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:190]: Correct me if I'm wrong here, but I like this will produce a button inside a link. It looks like it should just be a link

File: src/applications/vaos/appointment-list/components/cards/confirmed/AppointmentInstructions.jsx [hunk 20]
```diff
@@ -9,7 +12,14 @@ export default function AppointmentInstructions({ instructions }) {
...
   return (
     <div className="vads-u-flex--1 vads-u-margin-bottom--2 vaos-u-word-break--break-word">
-      <h4 className="vaos-appts__block-label">{header}</h4>
+      {isHomepageRefresh && (
+        <h2 className="vaos-appts__block-label vads-u-font-size--h4">
+          {header}
+        </h2>
+      )}
+      {!isHomepageRefresh && (
+        <h4 className="vaos-appts__block-label">{header}</h4>
```
Comment (timwright12) [inline src/applications/vaos/appointment-list/components/cards/confirmed/AppointmentInstructions.jsx:21]: What's the reason that this element is an h4, but the other version (line 16) is an h2? Seems like they should be the same heading level

File: src/applications/vaos/components/FacilityAddress.jsx [hunk 21]
```diff
@@ -37,17 +39,27 @@ export default function FacilityAddress({
...
+          </div>
+        )}
+      {!isV2 &&
+        !!phone && (
+          <dl className="vads-u-margin-y--0">
+            <dt className="vads-u-display--inline">
+              <strong>Main phone:</strong>
+            </dt>{' '}
+            <dd className="vads-u-display--inline">
+              <br />
```
Comment (timwright12) [inline src/applications/vaos/components/FacilityAddress.jsx:58]: If this is deliberate for spacing, can we use a class instead?

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 18 above)
Comment (jbalboni) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:160]: This functionality is in a later ticket, it's a placeholder for now

File: src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx:5]: No. I copied the current version of master to my branch to resolve some merge conflicts.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:516]: I figured there had to be a better way to do this. I just didn't know how. I need a tutorial on how to used renderWithStoreAndRouter and renderFromRoutes.

File: src/applications/vaos/appointment-list/components/cards/confirmed/AppointmentInstructions.jsx (hunk 20 above)
Comment (jbalboni) [reply src/applications/vaos/appointment-list/components/cards/confirmed/AppointmentInstructions.jsx:21]: These components are used on different pages, which different headings, so the level needs to fit within the heading order of the page it's being used on.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:516]: Done.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx (hunk 3 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:446]: Done.

File: src/applications/vaos/components/VAFacilityLocation.jsx (hunk 4 above)
Comment (vbahinwillit) [reply src/applications/vaos/components/VAFacilityLocation.jsx:10]: Fixed.

File: src/applications/vaos/appointment-list/redux/actions.js (hunk 5 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/actions.js:457]: Removed.

File: src/applications/vaos/components/FacilityAddress.jsx (hunk 6 above)
Comment (vbahinwillit) [reply src/applications/vaos/components/FacilityAddress.jsx:10]: Fixed.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 8 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:77]: Fixed.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 9 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:78]: Need more details.

File: src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx (hunk 10 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:34]: Fixed.

File: src/applications/vaos/appointment-list/index.jsx (hunk 11 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/index.jsx:29]: Fixed.

File: src/applications/vaos/appointment-list/redux/reducer.js (hunk 12 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/reducer.js:164]: Fixed.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx (hunk 13 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:546]: Cool. It wasn't clear to me what was causing the strange behavior I was seeing.

File: src/applications/vaos/appointment-list/redux/actions.js (hunk 15 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/actions.js:448]: Fixed.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx (hunk 16 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/AppointmentsPageV2.unit.spec.jsx:501]: Done.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 17 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:111]: Thanks for the review. I changed <b> to <strong> and removed the font weight style. I will check with are UI/UX person about whether his intent was for a header or bold text.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 19 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:190]: You are correct. I modified to style the link as a button.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 17 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:111]: It was decided to leave as is.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 7 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:52]: Modified to go back to the root route.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 14 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:79]: I'm not sure I understand completely but does creating another service or modifying the current getBookedAppointments service (flag or something) to only get VA confirmed appointments work?

File: src/applications/vaos/components/FacilityAddress.jsx (hunk 21 above)
Comment (vbahinwillit) [reply src/applications/vaos/components/FacilityAddress.jsx:58]: @jbalboni - What's the best way to address this?

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 14 above)
Comment (jbalboni) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:79]: I don't think you need to modify anything in the services, I think that you can assume that this page will never need to render an appointment that's community care. We can make sure the list pages link to a different page than this one when the appointment is CC.

File: src/applications/vaos/components/FacilityAddress.jsx (hunk 21 above)
Comment (jbalboni) [reply src/applications/vaos/components/FacilityAddress.jsx:58]: It's sort of in a weird spot, but it's not for spacing, it's to create a line break between the label and the content here.

I think I'd rather leave this the way it is for now, if we want to change it I'd want to look into our other uses of `dl` elements and why we've marked this one as an inline element, too. I can make a ticket for that.

File: src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx (hunk 14 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage.jsx:79]: Ok

**Review by timwright12 — APPROVED** [review]

File: src/applications/vaos/components/FacilityAddress.jsx (hunk 21 above)
Comment (vbahinwillit) [reply src/applications/vaos/components/FacilityAddress.jsx:58]: Thanks.

File: src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx [hunk 22]
```diff
@@ -1,186 +1,239 @@
...
-      state: 'WY',
-      address1: '2360 East Pershing Boulevard',
-    },
-  },
-  phone: {
-    main: '307-778-7550',
-  },
-};
-const var983GC = transformFacility(facility);
+import { getVAFacilityMock, getVARequestMock } from '../../mocks/v0';
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx:7]: @vbahinwillit I don't think I understand the changes happening in here. Are they necessary? cc @narin

Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx:7]: These changes are what's currently in the master branch.

Comment (vbahinwillit) [reply src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx:7]: Just ran the following:

git checkout master -- src/applications/vaos/tests/appointment-list/components/RequestedAppointmentDetailsPage.unit.spec.jsx

to insure that I'm not introducing anything new into the file. I moved my original test to a new file 'ConfirmedAppointmentDetailsPage.unit.spec.jsx'.

**Review by jbalboni — APPROVED** [review]
LGTM, though you need to pull in the latest from master to get 3 of those failing tests to pass
